### ISSUE TITLE
Trying to commit and getting Exit status "FastlaneCore::Interface::FastlaneShellError"

### ISSUE THREAD

**Issue by OfirKa** [fastlane/fastlane#11959]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Hello, 
I'm facing some issues while trying to add, commit and push new files to git. 
This is a short version of my code:

```
lane :Version_Upload do
... #testing code lined
...
git_add(path: ".")
UI.message "Please enter commit message"
commitMess = gets
git_commit(path:".", message: commitMess.to_s)
push_to_git_remote( local_branch: branchName)
end

```

At this point I am facing to main issues:

1) git_add does not add any files to be commited.
While trying to test my code, I've put a breakpoint right after the "git_add" command, and typed "git status" in my terminal.
At that point I still saw my changes in red color, and only after I manually typed "git add ." in the terminal I saw the files in green color.

2) **and this is the main issue** - 
I'm simply not able to commit with git_commit function.
After the code reaches git_commit line i'm getting the following error:

> Exit status of command 'git commit -m <Message i've typed> .' was 1 instead of 0.
> On branch <My Branch Name>
> Changes not staged for commit:
> <list of files that has been modified>
> no changes added to commit.

At this point fastlane crush. I'm getting the fastlane summery is displayed and at the end of it I get the following error:

> Exit status of command 'git commit -m <Message i've typed> .' was 1 instead of 0. (FastlaneCore::Interface::FastlaneShellError)

> from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane_core/lib/fastlane_core/ui/ui.rb:16:in `method_missing'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/helper/sh_helper.rb:80:in `sh_control_output'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/helper/sh_helper.rb:12:in `sh'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/actions/git_commit.rb:11:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
> 	from Fastfile:153:in `block (2 levels) in parsing_binding'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane.rb:33:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane.rb:33:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/lane_manager.rb:57:in `cruise_lane'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
> 	from /Users/userA/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.83.0/bin/fastlane:20:in `<top (required)>'

I've been trying to change the "path" and the "message" parameter in git_commit and still couldn't make it works.
What am I missing?

Thank you
